theano/gof/link.py

```diff
@@ -504,11 +507,13 @@ def map_storage(fgraph, order, input_storage, output_storage):
 #             raise TypeError("Cannot link a graph with non-constant orphans.", orphan)
 #         storage_map[orphan] = [orphan.data]
 
+    # allocate output storage
     if output_storage is not None:
```

Comment:
We have the same problem here.

I'm not sure what we should do. Do you have an idea? If we provide output_storage or input_storage and storage_map, do they need to match or we overide the content of the storage_map? Whatever the final version is, it must be in the docstring.